svc-systems/chassis/fabric.go

```diff
@@ -0,0 +1,259 @@
+//(C) Copyright [2020] Hewlett Packard Enterprise Development LP
+//
+//Licensed under the Apache License, Version 2.0 (the "License"); you may
+//not use this file except in compliance with the License. You may obtain
+//a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//Unless required by applicable law or agreed to in writing, software
+//distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+//WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+//License for the specific language governing permissions and limitations
+// under the License.
+
+package chassis
+
+import (
+	"encoding/json"
+	"fmt"
+	dmtf "github.com/ODIM-Project/ODIM/lib-dmtf/model"
+	"github.com/ODIM-Project/ODIM/lib-rest-client/pmbhandle"
+	"github.com/ODIM-Project/ODIM/lib-utilities/common"
+	"github.com/ODIM-Project/ODIM/lib-utilities/config"
+	"github.com/ODIM-Project/ODIM/svc-systems/smodel"
+	"github.com/ODIM-Project/ODIM/svc-systems/sresponse"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+type fabricFactory struct {
+	collection        *sresponse.Collection
+	chassisMap        map[string]bool
+	wg                *sync.WaitGroup
+	mu                *sync.Mutex
+	getFabricManagers func() ([]smodel.Plugin, error)
+	contactClient     func(string, string, string, string, interface{}, map[string]string) (*http.Response, error)
+}
+
+func getFabricFactory(collection *sresponse.Collection) *fabricFactory {
+	chassisMap := make(map[string]bool)
+	return &fabricFactory{
+		collection:        collection,
+		chassisMap:        chassisMap,
+		wg:                &sync.WaitGroup{},
+		mu:                &sync.Mutex{},
+		getFabricManagers: smodel.GetFabricManagers,
+		contactClient:     pmbhandle.ContactPlugin,
+	}
+}
+
+type pluginContactRequest struct {
+	URL             string
+	HTTPMethodType  string
+	ContactClient   func(string, string, string, string, interface{}, map[string]string) (*http.Response, error)
+	PostBody        interface{}
+	LoginCredential map[string]string
+	Plugin          smodel.Plugin
+	Token           string
+}
+
+// PluginToken interface to hold the token
+type PluginToken struct {
+	Tokens map[string]string
+	lock   sync.Mutex
+}
+
+// Token variable hold the all the XAuthToken  against the plguin ID
+var Token PluginToken
+
+func (c *sourceProviderImpl) findFabricChassis(collection *sresponse.Collection) {
+	f := c.getFabricFactory(collection)
+	managers, err := f.getFabricManagers()
+	if err != nil {
+		log.Warn("while trying to collect fabric managers details from DB, got " + err.Error())
+		return
+	}
+	for _, manager := range managers {
+		f.wg.Add(1)
+		go f.getFabricCollection(manager)
+	}
+	f.wg.Wait()
+}
+
+func (f *fabricFactory) getFabricCollection(plugin smodel.Plugin) {
```

Comment:
rename the method  this method not collecting fabric information

Reply:
addressed